### PR TITLE
fix: Adding support for Cilium Network Policy

### PR DESCRIPTION
**Pull request by cv-gh**
# Supporting Cilium network policy in AKS

cilium data plane requires cilium network policy. when specifying azure policy for cilium we get an error - CiliumDataplaneRequiresNetworkPolicyCilium

Configuration
networkPolicy: 'azure'
networkPlugin: 'azure'
networkDataplane: 'cilium'

Detailed Error
```json
{
    "message": {
        "code": "BadRequest",
        "details": null,
        "message": "Cilium dataplane requires network policy cilium.",
        "subcode": "CiliumDataplaneRequiresNetworkPolicyCilium",
        "target": "networkProfile.networkPolicy"
    }
}
```

| Pipeline |
| -------- |
|      [avm.res.container-service.managed-cluster](https://github.com/cv-gh/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)    |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ x] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ x] I'm sure there are no other open Pull Requests for the same update/change
- [ x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by ReneHezser** [comment]
@cv-gh 
We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.

Please reach out if any support is needed.

**Comment by ReneHezser** [comment]
> Hi @ReneHezser set-avmmodule PowerShell was executed and readme and main.json are included in the change. static code validation is passing but build issue is due to some other issues unrelated to this PR. Issue is about AKS Schedule update configuration, and I think there is another PR #3228

Hi @cv-gh 
You need to update your branch first. Only then will the new script version update the Readme.

**Comment by ilhaan** [comment]
Looks like https://github.com/Azure/bicep-registry-modules/pull/3228 has merged. @cv-gh can you please merge main into your branch and re-run tests?

**Comment by eriqua** [comment]
> Looks like #3228 has merged. @cv-gh can you please merge main into your branch and re-run tests?

Hi @cv-gh,
Echoing @ilhaan, please merge latest from upstream. 
The PR shows you currently have conflicts to be solved in main.json, hence it cannot be approved and merged.
Please reach out if you need any support.
Let's drive this useful contribution to closure 💪

File: avm/res/container-service/managed-cluster/README.md
Comment (AlexanderSehr) [inline avm/res/container-service/managed-cluster/README.md:1]: Please note, the `main.json` templates **must** be re-build with the latest Bicep version `0.31.34` as this version (running on the agents) renders a different `main.json` file then previous versions. 

The steps to execute would be
- Upgrade your Bicep CLI version to `0.31.34` (e.g. via `choco upgrade bicep` if choco is used)
- Re-run the `Set-AVMModule` script on your module. E.g. via `Set-AVMModule -ModuleFolderPath '.\avm\res\automation\automation-account\' -SkipFileAndFolderSetup -SkipReadMe -Recurse`
- Push the changes
- Re-run the tests to ensure everything works. 

If the static tests continue to fail for you, please note that there is currently presumably another new bug which affects some of our modules ([ref](https://github.com/Azure/bicep/issues/15517)).

Comment (AlexanderSehr) [reply avm/res/container-service/managed-cluster/README.md:1]: Hey @cv-gh,
that's true - but it's fortunately an easy fix as the test is just complaining that the agent pool's `main.json` is still in an outdated state: 
image

In this case you're simply exposed to a non-breaking change that was introduced by the latest Bicep release and effects us on all levels of a module (e.g., the parent module, as well as the child-modules like in this case `agent-pool`). You should be good as soon as you also build that file. Or - like in the original comment, you simply run the `Set-AVMModule` script with the `-Recurse` parameter.

**Review by JPEasier — APPROVED** [review]
Looks good for me 🚀

**Pull request merged**